.github/workflows/docs.yml

```diff
@@ -35,5 +35,39 @@ jobs:
       - name: Upload docs as GHA artifact
         uses: actions/upload-artifact@v2
         with:
-          name: docs.whl
+          name: built-docs
           path: docs/build/html
+
+  deploy-docs:
+    needs: [build-docs]
+    runs-on: ubuntu-latest
+    if: (github.ref == 'refs/heads/main' && github.event_name == 'push')
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          ref: gh-pages
+          fetch-depth: 3
+
+      - name: Download docs artifact
+        uses: actions/download-artifact@v2
+        with:
+          name: built-docs
+          path: /tmp/docs
+
+      - name: Copy built docs to nightly
+        run: |
+          cp -R /tmp/docs/* nightly/
+          # Manipulations to squash the history:
+          # 1) reset to a fixed commit
+          # 2) use github-actions-x/commit to commit everything as a single commit
+          git reset --soft 218f32faf8b8cc5236bc49ea77ef7d222e21e5d6
```

Comment:
I thought about it some more, and I think it would be better if we didn't reset the history for now (but turned it on later). The reasoning is:
1. We might do manual commits to the docs from time-to-time
2. We don't actually produce that much docs, so it'll take a while (a couple of months?) before we actually run into memory problems

What are your thoughts?

Reply:
Hmm, I like the proposal. Ideally I want some way to be able to roll the website back to a "last known good configuration" if we accidentally merge a bad change and are unable to fix it quickly.

One possible solution for that is: every week, on Sunday, have github actions do a manual commit to the gh-pages branch. The net effect of this is that we get roughly one new docs commit per week, and if something goes bad during the week, then we can roll the website back to its state from last week.

Reply:
Sounds good about cron-based GHA.

How about trying to make commit amend if previous commit was auto-generated and add a new commit otherwise ? In this case, we can safely roll back if needed and do not accumulated many auto-generated commits...
+ we can do everything cron-based (maybe daily).

Reply:
> How about trying to make commit amend if previous commit was auto-generated and add a new commit otherwise ? In this case, we can safely roll back if needed and do not accumulated many auto-generated commits... And we can do everything cron-based (maybe daily).

Okay! That sounds good. Hopefully we have not overengineered the system too much :)